### PR TITLE
Add envelope wrapper pattern sample

### LINKED ISSUE #71: Create Sample for Envelop Wrapper
**Description:**
Create sample for [Envelope Wrapper](https://www.enterpriseintegrationpatterns.com/patterns/messaging/EnvelopeWrapper.html)

### PR DESCRIPTION
**Pull request by prakanth97**
## Purpose
Fixes https://github.com/ballerina-guides/enterprise-integration-patterns/issues/71

### CODE REVIEW

File: envelope-wrapper/envelope-wrapper.bal [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+service /paypal on new http:Listener(8080) {
+    resource function post refund(RefundRequest refundReq) returns PaypalRespose|error {
+        // Generate PayPal-Auth-Assertion.
+        string authAssertionValue = getAuthAssertionValue(refundReq.client_id, refundReq.email);
+
+        http:Request request = new;
+        request.addHeader("PayPal-Request-Id", refundReq.request_id);
+        request.addHeader("PayPal-Auth-Assertion", authAssertionValue);
+        return paypalClient->/v2/payments/captures/[refundReq.capture_id]/refund.post(request);
```
Comment (LakshanWeerasinghe) [inline envelope-wrapper/envelope-wrapper.bal:26]: Is it correct to send a post request with an empty body?

File: envelope-wrapper/envelope-wrapper.bal [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    }
+}
+
+isolated function getAuthAssertionValue(string client_id, string email) returns string {
+    string encodeHeader = {alg: "none"}.toString().toBytes().toBase64();
```
Comment (manuranga) [inline envelope-wrapper/envelope-wrapper.bal]: Let's have in a module-level final variable.

File: envelope-wrapper/envelope-wrapper.bal [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    map<string> payload = {
+        iss: client_id,
+        payer_id: email
+    };
+    string encodePayload = payload.toString().toBytes().toBase64();
+
```
Comment (manuranga) [inline envelope-wrapper/envelope-wrapper.bal]: get rid of this empty line.

File: envelope-wrapper/envelope-wrapper.bal (hunk 1 above)
Comment (prakanth97) [reply envelope-wrapper/envelope-wrapper.bal:26]: Updated

File: envelope-wrapper/envelope-wrapper.bal [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+service /paypal on new http:Listener(8080) {
+    resource function post refund(RefundRequest refundReq) returns PaypalRespose|error {
+        // Generate PayPal-Auth-Assertion.
+        string authAssertionValue = getAuthAssertionValue(refundReq.client_id, refundReq.email);
+
+        http:Request request = new;
+        request.addHeader("PayPal-Request-Id", refundReq.request_id);
+        request.addHeader("PayPal-Auth-Assertion", authAssertionValue);
+        request.setPayload({});
```
Comment (manuranga) [inline envelope-wrapper/envelope-wrapper.bal]: no need.

**Review by manuranga — APPROVED** [review]

**Pull request merged**